sql/log.cc

```diff
@@ -7970,6 +7972,20 @@ MYSQL_BIN_LOG::queue_for_group_commit(group_commit_entry *orig_entry)
     DBUG_ASSERT(entry != NULL);
     cur= entry->thd->wait_for_commit_ptr;
   }
+#ifdef WITH_WSREP
+  if (wsrep_is_active(entry->thd) &&
+      wsrep_run_commit_hook(entry->thd, entry->all))
+  {
+    /*  Release commit order here */
+    if (wsrep_ordered_commit(entry->thd, entry->all))
+    {
+      result= -2;
+    }
+    /* return -3, if this is leader */
+    if (orig_queue == NULL)
+        result= -3;
+  }
+#endif /* WITH_WSREP */
```

Comment:
I think that this must be reviewed by @andrelkin. What is the overhead of the `wsrep_is_active()` call when Galera replication is not enabled? It looks like it might be OK, if `MYSQL_BIN_LOG::write_transaction_to_binlog_events()` is the only caller of `MYSQL_BIN_LOG::queue_for_group_commit()`, because a call was removed from there.

Formatting: Why is `result= -2` surrounded by redundant `{` and `}`? Why is `result= -3` indented deeper?

Where are the debug assertions for checking that `-2` and `-3` are never returned in other cases?